Source/NonlinearSolvers/NewtonSolver.H

```diff
@@ -212,6 +218,9 @@ void NewtonSolver<Vec,Ops>::ParseParameters ()
     pp_gmres.query("absolute_tolerance",  m_gmres_atol);
     pp_gmres.query("relative_tolerance",  m_gmres_rtol);
     pp_gmres.query("max_iterations",      m_gmres_maxits);
+
+    const amrex::ParmParse pp_jac("jacobian");
+    pp_jac.query("pc_type", m_pc_type);
```

Comment:
We could make m_pc_type an enum type instead of string with AMREX_NUM.
```
AMREX_ENUM(PreconditionerType, pc_curl_curl_mlmg, none);
```
Then `std::string m_pc_type = "none";` becomes `auto m_pc_type = PreconditionerType::none;`, and you also don't need
```
namespace PreconditionerTypes
{   
    /**                                                                                                                                                                      
     * MLMG-based solver for the curl-curl operator                                                                                                                          
     */
    const std::string curl_curl_mlmg = "pc_curl_curl_mlmg";
}
```
When you need a string, you can turn the enum into a string with `std::string s = amrex::getEnumNameString(m_pc_type);`.